src/structures/gdt.rs

```diff
@@ -105,6 +105,25 @@ impl GlobalDescriptorTable {
         }
     }
 
+    /// Forms a GDT from a raw table and a next_free counter.
+    ///
+    /// # Safety
+    ///
+    /// The user must ensure that the provided table contains well formed entries
+    /// and that the value of `next_free` does not exceed the maximum size of the table.
+    #[inline]
+    pub const unsafe fn from_raw_parts(table: [u64; 8], next_free: usize) -> GlobalDescriptorTable {
+        GlobalDescriptorTable { table, next_free }
+    }
+
+    /// Breaks a GDT into its raw parts (table and a next_free counter.)
+    #[inline]
+    pub const fn into_raw_parts(&self) -> ([u64; 8], usize) {
+        let Self { table, next_free } = *self;
+
+        (table, next_free)
+    }
```

Comment:
I agree with the clippy suggestion that the `into_*` method name suggests that the method takes `self` by value. We should rename the method to e.g. ` as_raw_parts` or `raw_parts` (or `[as_]raw_slice`).